Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h

```diff
@@ -97,7 +105,7 @@ class TrackerGeometry final : public TrackingGeometry {
 
   GeomDetEnumerators::SubDetector theSubDetTypeMap[6];
   unsigned int theNumberOfLayers[6];
-
+  std::vector< std::tuple< DetId, TrackerGeometry::ModuleType, float> > theDetTypetList; 
```

Comment:
I think that the use of the db at this stage would be an overkill. This vector is filled once at the beginning from the geometry (which is presently in xml only).